session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -1790,6 +1790,81 @@ public void createSchemaTemplate(
     defaultSessionConnection.createSchemaTemplate(request);
   }
 
+  public void createSchemaTemplate(String templateName, List<IMeasurementSchema> measurementSchemas)
+      throws IoTDBConnectionException, StatementExecutionException {
+    List<String> schemaNames = new ArrayList<>();
+    List<List<String>> measurements = new ArrayList<>();
+    List<List<TSDataType>> dataTypes = new ArrayList<>();
+    List<List<TSEncoding>> encodings = new ArrayList<>();
+
+    List<CompressionType> compressors = new ArrayList<>();
+    for (IMeasurementSchema measurementSchema : measurementSchemas) {
+      schemaNames.add(measurementSchema.getMeasurementId());
+      measurements.add(Collections.singletonList(measurementSchema.getMeasurementId()));
+      dataTypes.add(Collections.singletonList(measurementSchema.getType()));
+      encodings.add(Collections.singletonList(measurementSchema.getEncodingType()));
```

Comment:
When the measurementSchemas is a VectorMeasurementSchema, the `Collections.singletonList` seems not correct.

Reply:
edited as required